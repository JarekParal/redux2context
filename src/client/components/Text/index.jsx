// @flow strict
/* eslint-disable react/no-danger */
import * as React from "react";

import { Consumer } from "client/services/intl/context";

// TODO add actual translation
type Props = {
  t: string,
  // values?: { [key: string]: string | number },
  html: boolean,
};

const Text = (props: Props) => (
  <Consumer>
    {intl =>
      props.html ? (
        <span dangerouslySetInnerHTML={{ __html: intl.translations[props.t] || props.t }} />
      ) : (
        intl.translations[props.t] || props.t
      )
    }
  </Consumer>
);

Text.defaultProps = {
  // values: {},
  html: false,
};

export default Text;
