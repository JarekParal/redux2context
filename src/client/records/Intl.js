// @flow strict
export type IntlRaw = {|
  locale: string,
  translations: { [key: string]: string },
|};

export type Intl = {|
  ...IntlRaw,
  t: (key: string, values?: { [key: string]: string }) => string,
|};

// eslint-disable-next-line import/prefer-default-export
export const intlDefault: Intl = {
  locale: "en",
  translations: {},
  t: id => id,
};
