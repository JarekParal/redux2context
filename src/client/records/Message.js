// @flow strict
export type Message = {|
  id: string,
  from: string,
  text: string,
|};

export type Messages = { [id: string]: Message };
