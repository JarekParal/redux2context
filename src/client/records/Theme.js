// @flow strict
export type Theme = {|
  name: string,
  colors: {|
    primary: string,
  |},
|};

export type ThemeProps = {
  theme: Theme,
};

// eslint-disable-next-line import/prefer-default-export
export const themeDefault: Theme = {
  name: "Reactizer",
  colors: {
    primary: "deepskyblue",
  },
};
