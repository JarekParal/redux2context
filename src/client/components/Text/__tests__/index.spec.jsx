// @flow strict
import * as React from "react";
import { shallow } from "enzyme";

import Text from "../index";
import { intlDefault } from "../../../records/Intl";

describe("#Text", () => {
  test("render", () => {
    const wrapper = shallow(<Text t="lol" />);

    expect(wrapper.prop("children")(intlDefault)).toMatchSnapshot();
  });

  test("render as span", () => {
    const wrapper = shallow(<Text t="lol" html />);

    expect(wrapper.prop("children")(intlDefault)).toMatchSnapshot();
  });
});
