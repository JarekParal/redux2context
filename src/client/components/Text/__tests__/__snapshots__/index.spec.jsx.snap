// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Text render 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`#Text render as span 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
